docs/resource_sets.rst

```diff
@@ -91,3 +93,291 @@ Limitations
+
+In this guide, we only cover models where each set of independent resources is defined by a single top-level
+entity, which we will refer to as the "service" or "service entity" (as in ``LSM``).
+
+When building an inmanta model all potential instances of a service entity must refine to compatible (low level) config.
+In the model this config is represented by the resources. Therefore these guidelines will focus on creating valid and
+compatible resources. With well-designed resources, valid and compatible config will follow.
+
+To safely make use of partial compiles, each service must be the sole owner of its resources and any shared resources must be
+identical across service instances. The graphs below picture respectively a valid and an invalid service for partial compiles.
+Each arrow represents a refinement: one entity creating another in one of its implementations. The valid service results in
+fully separate resource sets for each instance, while the invalid one has resources that overlap
+between instances. Additionally, the shared resource for the valid service is consistent, but the invalid service attempts to
+create a resource with the same id, yet a different value. The invalid service can thus not be allowed for partial compiles
+because of the inconsistency of shared resources and the fact that no resource can be considered completely owned by a single
+service instance.
+
+.. digraph:: resource_sets_good
+    :caption: A good service for partial compiles.
+
+    subgraph cluster_service_good0 {
+        "GoodService(id=0)" [shape=rectangle];
+        "GoodService(id=0)" -> subgraph cluster_resources_good0 {
+            "Resource(id=0)";
+            "Resource(id=1)";
+            color = "grey";
+            fontcolor = "grey";
+            label = "Resource set for GoodService(id=0)";
+            labelloc = "bottom";
+        }
+        label = "Owned by GoodService(id=0)";
+        labelloc = "top";
+        color="green";
+    }
+    subgraph cluster_service_good1 {
+        "GoodService(id=1)" [shape=rectangle];
+        "GoodService(id=1)" -> subgraph cluster_resources_good1 {
+            "Resource(id=2)";
+            "Resource(id=3)";
+            color = "grey";
+            fontcolor = "grey";
+            label = "Resource set for GoodService(id=1)";
+            labelloc = "bottom";
+        }
+        label = "Owned by GoodService(id=1)";
+        labelloc = "top";
+        color="green";
+    }
+    { "GoodService(id=0)" "GoodService(id=1)" } -> subgraph cluster_resources_good_shared {
+        "SharedResource(id=0, value=0)";
+        color="green";
+        fontcolor="grey";
+        label = "Shared and consistent among all service instances";
+        labelloc = "bottom";
+    }
+
+    # force rendering on multiple ranks
+    {"Resource(id=0)" "Resource(id=1)" "Resource(id=2)" "Resource(id=3)"} -> "SharedResource(id=0, value=0)" [style="invis"]
+
+
+.. digraph:: resource_sets_generic_bad
```

Comment:
I would add the explanation of the bad one after the good one, this has a more natural flow.

Reply:
It may also help to split both bad things into their own example

Reply:
Done